sw/device/tests/dif/dif_uart_test.cc

```diff
@@ -0,0 +1,645 @@
+                     {UART_FIFO_CTRL_RXRST, true}, {UART_FIFO_CTRL_TXRST, true},
+                 });
+
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET, 0);
+
+  EXPECT_TRUE(dif_uart_configure(&dif_uart_, &dif_uart_config_));
+}
+
+TEST_F(ConfigTest, ParityEven) {
+  ExpectDeviceReset();
+
+  dif_uart_config_.parity_enable = kDifUartEnable;
+  dif_uart_config_.parity = kDifUartParityEven;
+
+  EXPECT_WRITE32(UART_CTRL_REG_OFFSET, {
+                                           {UART_CTRL_TX, true},
+                                           {UART_CTRL_RX, true},
+                                           {UART_CTRL_PARITY_EN, true},
+                                           {UART_CTRL_NCO_OFFSET, 1},
+                                       });
+
+  EXPECT_WRITE32(UART_FIFO_CTRL_REG_OFFSET,
+                 {
+                     {UART_FIFO_CTRL_RXRST, true}, {UART_FIFO_CTRL_TXRST, true},
+                 });
+
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET, 0);
+
+  EXPECT_TRUE(dif_uart_configure(&dif_uart_, &dif_uart_config_));
+}
+
+TEST_F(ConfigTest, ParityOdd) {
+  ExpectDeviceReset();
+
+  dif_uart_config_.parity_enable = kDifUartEnable;
+  dif_uart_config_.parity = kDifUartParityOdd;
+
+  EXPECT_WRITE32(UART_CTRL_REG_OFFSET, {
+                                           {UART_CTRL_TX, true},
+                                           {UART_CTRL_RX, true},
+                                           {UART_CTRL_PARITY_EN, true},
+                                           {UART_CTRL_PARITY_ODD, true},
+                                           {UART_CTRL_NCO_OFFSET, 1},
+                                       });
+
+  EXPECT_WRITE32(UART_FIFO_CTRL_REG_OFFSET,
+                 {
+                     {UART_FIFO_CTRL_RXRST, true}, {UART_FIFO_CTRL_TXRST, true},
+                 });
+
+  EXPECT_WRITE32(UART_INTR_ENABLE_REG_OFFSET, 0);
+
+  EXPECT_TRUE(dif_uart_configure(&dif_uart_, &dif_uart_config_));
+}
+
+class WatermarkRxSetTest : public UartTest {};
+
+TEST_F(WatermarkRxSetTest, UartNull) {
+  EXPECT_FALSE(dif_uart_watermark_rx_set(nullptr, kDifUartWatermarkByte1));
+}
+
+/**
+ * Tests the first and the last RX watermark variants.
+ */
+TEST_F(WatermarkRxSetTest, Success) {
+  EXPECT_MASK32(UART_FIFO_CTRL_REG_OFFSET,
+                {
+                    {UART_FIFO_CTRL_RXILVL_OFFSET, UART_FIFO_CTRL_RXILVL_MASK,
+                     UART_FIFO_CTRL_RXILVL_RXLVL1},
+                });
+
+  EXPECT_TRUE(dif_uart_watermark_rx_set(&dif_uart_, kDifUartWatermarkByte1));
+
+  EXPECT_MASK32(UART_FIFO_CTRL_REG_OFFSET,
+                {
+                    {UART_FIFO_CTRL_RXILVL_OFFSET, UART_FIFO_CTRL_RXILVL_MASK,
+                     UART_FIFO_CTRL_RXILVL_RXLVL30},
+                });
+
+  EXPECT_TRUE(dif_uart_watermark_rx_set(&dif_uart_, kDifUartWatermarkByte30));
+}
+
+class WatermarkTxSetTest : public UartTest {};
+
+TEST_F(WatermarkTxSetTest, UartNull) {
+  EXPECT_FALSE(dif_uart_watermark_tx_set(nullptr, kDifUartWatermarkByte1));
```

Comment:
You should have a test here for the error condition associated with `kDifUartWatermarkByte30`. Here I wrote it for you:
```c
  EXPECT_FALSE(dif_uart_watermark_tx_set(&dif_uart_, kDifUartWatermarkByte30));
```

Reply:
Good point, I will add it.